crates/wasmtime/src/runtime/module.rs

```diff
@@ -1133,14 +1143,18 @@ impl Module {
     /// Get the text offset (relative PC) for a given absolute PC in
     /// this module.
     #[cfg(any(feature = "gc", feature = "debug"))]
-    pub(crate) fn text_offset(&self, pc: usize) -> u32 {
-        u32::try_from(pc - self.inner.module.text().as_ptr() as usize).unwrap()
+    pub(crate) fn text_offset(&self, pc: usize, code_memory: &CodeMemory) -> u32 {
```

Comment:
Since this doesn't actually use `&self` at all here any more, maybe this could move to `CodeMemory` directly?

Reply:
(Updated after refactor)